src/app/search-page/filter-component/filter.component.html

```diff
@@ -0,0 +1,30 @@
+<div *ngIf="this.query">
```

Comment:
I think the code would be easier to understand if you would move the *ngIf="this.query" to the parent component.
Like this you could remove the query from this component.

Reply:
you are right! I moved it to the parent component